rs/nns/sns-wasm/src/sns_wasm.rs

```diff
@@ -480,18 +480,21 @@ where
                     },
                 );
 
-                self.upgrade_path.add_wasm(sns_canister_type, &hash);
-
-                Some(add_wasm_response::Result::Hash(hash.to_vec()))
+                if let Err(err) = self.upgrade_path.add_wasm(sns_canister_type, &hash) {
```

Comment:
This is failing after we've already written it into memory.  It would be better to do a check before writing the wasm ( a few lines above) to see if the upgrade_path already includes it.

Reply:
Just to be clear, please fix this before merging, so we're not unnecessarily writing to stable memory and then rejecting the update afterwards.

Reply:
Fixed!